saga/seata-saga-engine/src/main/java/io/seata/saga/engine/impl/ProcessCtrlStateMachineEngine.java

```diff
@@ -0,0 +1,633 @@
+            throws EngineExecutionException {
+
+        return startInternal(stateMachineName, tenantId, null, startParams, true, callback);
+    }
+
+    @Override
+    public StateMachineInstance startWithBusinessKey(String stateMachineName, String tenantId, String businessKey, Map<String, Object> startParams)
+            throws EngineExecutionException {
+
+        return startInternal(stateMachineName, tenantId, businessKey, startParams, false, null);
+    }
+
+    @Override
+    public StateMachineInstance startWithBusinessKeyAsync(String stateMachineName, String tenantId, String businessKey, Map<String, Object> startParams,
+                                           AsyncCallback callback)
+            throws EngineExecutionException {
+
+        return startInternal(stateMachineName, tenantId, businessKey, startParams, true, callback);
+    }
+
+    private StateMachineInstance startInternal(String stateMachineName, String tenantId, String businessKey, Map<String, Object> startParams, boolean async, AsyncCallback callback)
+            throws EngineExecutionException {
+
+        if (async && !stateMachineConfig.isEnableAsync()) {
+            throw new EngineExecutionException("Asynchronous start is disabled. please set StateMachineConfig.enableAsync=true first.",
+                    FrameworkErrorCode.AsynchronousStartDisabled);
+        }
+
+        if(StringUtils.isEmpty(tenantId)){
+            tenantId = stateMachineConfig.getDefaultTenantId();
+        }
+
+        StateMachineInstance instance = createMachineInstance(stateMachineName, tenantId, businessKey, startParams);
+
+        ProcessContextBuilder contextBuilder = ProcessContextBuilder.create().withProcessType(ProcessType.STATE_LANG)
+                .withOperationName(DomainConstants.OPERATION_NAME_START)
+                .withAsyncCallback(callback)
+                .withInstruction(new StateInstruction(stateMachineName, tenantId))
+                .withStateMachineInstance(instance)
+                .withStateMachineConfig(getStateMachineConfig())
+                .withStateMachineEngine(this);
+
+        Map<String, Object> contextVariables;
+        if(startParams != null){
+            contextVariables = new ConcurrentHashMap<>(startParams.size());
+            nullSafeCopy(startParams, contextVariables);
+        }
+        else{
+            contextVariables = new ConcurrentHashMap<>();
+        }
+        instance.setContext(contextVariables);
+
+        contextBuilder.withStateMachineContextVariables(contextVariables);
+
+        ProcessContext processContext = contextBuilder.build();
+
+        if (instance.getStateMachine().isPersist() && stateMachineConfig.getStateLogStore() != null) {
+            stateMachineConfig.getStateLogStore().recordStateMachineStarted(instance, processContext);
+        }
+        if(StringUtils.isEmpty(instance.getId())){
```

Comment:
The dbType property is designed for the future. DML has no special statements. In theory, it can support all databases. Currently tested mysql, db2, oracle, h2. We provide mysql, db2, oracle, h2 create schema scripts by default.